docs/src/main/asciidoc/resteasy-reactive-virtual-threads.adoc

```diff
@@ -0,0 +1,308 @@
+.pom.xml
+----
+<dependency>
+  <groupId>io.quarkus</groupId>
+  <artifactId>quarkus-resteasy-reactive-jackson</artifactId>
+</dependency>
+<dependency>
+  <groupId>io.quarkus</groupId>
+  <artifactId>quarkus-rest-client-reactive-jackson</artifactId>
+</dependency>
+----
+
+[source,gradle,role="secondary asciidoc-tabs-target-sync-gradle"]
+.build.gradle
+----
+implementation("io.quarkus:quarkus-resteasy-reactive-jackson")
+implementation("quarkus-rest-client-reactive-jackson")
+----
+
+[NOTE]
+====
+You may wonder why we use _reactive_ extensions.
+Virtual threads have limitations, and we can only integrate them properly when using the reactive extensions.
+However, no worries, your code will be 100% synchronous.
+
+Check the xref:./virtual-threads.adoc#why-not[virtual thread reference guide] for details.
+====
+
+== Prepare the `pom.xml` file
+
+We need to customize the `pom.xml` file to allow using virtual threads.
+
+1) Locate the line with `<maven.compiler.release>17</maven.compiler.release>`, and replace it with:
+
+[source, xml]
+----
+    <maven.compiler.release>20</maven.compiler.release>
+----
+
+2) In the maven-compiler-plugin configuration, add the `--enable-preview` arg (only if you use Java 19 or Java 20)
+
+[source, xml]
+----
+<plugin>
+  <artifactId>maven-compiler-plugin</artifactId>
+  <version>${compiler-plugin.version}</version>
+  <configuration>
+    <compilerArgs>
+      <arg>-parameters</arg>
+      <arg>--enable-preview</arg> <!-- Added line -->
+    </compilerArgs>
+  </configuration>
+</plugin>
+----
+
+3) In the maven-surefire-plugin and maven-failsafe-plugin configurations, add the following `argLine` parameter:
+
+[source, xml]
+----
+<plugin>
+  <artifactId>maven-surefire-plugin</artifactId>
+  <version>${surefire-plugin.version}</version>
+  <configuration>
+    <systemPropertyVariables>
+      <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
+      <maven.home>${maven.home}</maven.home>
+    </systemPropertyVariables>
+    <argLine>--enable-preview -Djdk.tracePinnedThreads</argLine> <!-- Added line -->
+  </configuration>
+</plugin>
+<plugin>
+  <artifactId>maven-failsafe-plugin</artifactId>
+  <version>${surefire-plugin.version}</version>
+  <executions>
+    <execution>
+      <goals>
+        <goal>integration-test</goal>
+        <goal>verify</goal>
+      </goals>
+      <configuration>
+        <systemPropertyVariables>
+          <native.image.path>${project.build.directory}/${project.build.finalName}-runner</native.image.path>
+          <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
+          <maven.home>${maven.home}</maven.home>
+        </systemPropertyVariables>
+        <argLine>--enable-preview -Djdk.tracePinnedThreads</argLine> <!-- Added line -->
+      </configuration>
+    </execution>
+  </executions>
+</plugin>
+----
+Note that the `--enable-preview` flag is only required if you use Java 19 or 20.
```

Comment:
I'd extract this with an `important` admonition and make sure this pops out as it's quite critical.